apollo-router/tests/common.rs

```diff
@@ -553,7 +552,9 @@ impl IntegrationTest {
                         level: String,
                         message: String,
                     }
-                    let log = serde_json::from_str::<Log>(&line).expect("line: '{line}' isn't JSON, might you have some debug output in the logging?");
+                    let Ok(log) = serde_json::from_str::<Log>(&line) else {
+                        panic!("line: '{line}' isn't JSON, might you have some debug output in the logging?");
+                    };
```

Comment:
This one was a legit bug! `{line}` was clearly intended as string formatting which `expect()` doesn’t do.

Reply:
Shouldn't there be an `if` somewhere? Can you use `else` on options/results now?

Reply:
No, this use of [let-else](https://doc.rust-lang.org/beta/rust-by-example/flow_control/let_else.html) is intended. `let` could originally only be used with irrefutable patterns, but was then extended to allow refutable patterns like `Ok(log)` if an `else` is provided that "diverges" (never reaches the end of the else block)

The difference with `if let` is that the `log` binding belong to the outer block

Reply:
Never mind, I looked at this in the playground and it's fine. TIL.